src/network/networkd-wwan.c

```diff
@@ -0,0 +1,444 @@
+        else
+                address_unmark(existing);
+
+        r = link_request_address(link, TAKE_PTR(address), true, &link->bearer_messages,
+                                 bearer_address_handler, NULL);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to request address provided by bearer: %m");
+
+        return 0;
+}
+
+static int bearer_route_handler(sd_netlink *rtnl, sd_netlink_message *m, Request *req, Link *link, Route *route) {
+        int r;
+
+        assert(link);
+
+        r = route_configure_handler_internal(rtnl, m, link, "Could not set gateway provided by bearer");
+        if (r <= 0)
+                return r;
+
+        if (link->bearer_messages == 0) {
+                link->bearer_configured = true;
+                link_check_ready(link);
+        }
+
+        return 0;
+}
+
+static int link_request_bearer_route(Link *link, int family, const union in_addr_union *gw, const union in_addr_union *prefsrc) {
+        _cleanup_(route_freep) Route *route = NULL;
+        Route *existing;
+        int r;
+
+        assert(link);
+        assert(IN_SET(family, AF_INET, AF_INET6));
+        assert(gw);
+        assert(prefsrc);
+
+        if (!in_addr_is_set(family, gw))
+                return 0;
+
+        r = route_new(&route);
+        if (r < 0)
+                return log_oom();
+
+        route->source = NETWORK_CONFIG_SOURCE_MODEM_MANAGER;
+        route->family = family;
+        route->gw_family = family;
+        route->gw = *gw;
+        if (prefsrc)
+                route->prefsrc = *prefsrc;
+
+        if (route_get(NULL, link, route, &existing) < 0) /* This is a new route. */
+                link->bearer_configured = false;
+        else
+                route_unmark(existing);
+
+        r = link_request_route(link, TAKE_PTR(route), true, &link->bearer_messages,
+                               bearer_route_handler, NULL);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to request gateway provided by bearer: %m");
+
+        return 0;
+}
+
+static int link_apply_bearer_impl(Link *link, Bearer *b) {
+        Address *address;
+        Route *route;
+        int r, ret = 0;
+
+        assert(link);
+
+        if (!IN_SET(link->state, LINK_STATE_CONFIGURING, LINK_STATE_CONFIGURED))
+                return 0;
+
+        /* First, mark bearer configs. */
+        SET_FOREACH(address, link->addresses) {
+                if (address->source != NETWORK_CONFIG_SOURCE_MODEM_MANAGER)
+                        continue;
+
+                address_mark(address);
+        }
+
+        SET_FOREACH(route, link->routes) {
+                if (route->source != NETWORK_CONFIG_SOURCE_MODEM_MANAGER)
+                        continue;
+
+                route_mark(route);
+        }
+
+        if (b && FLAGS_SET(b->ip_type, ADDRESS_FAMILY_IPV4)) {
+                if (b->connected && b->ip4_method == MM_BEARER_IP_METHOD_STATIC) {
```

Comment:
What about `MM_BEARER_IP_METHOD_PPP`? looks like that would be totally ignored, right? is it out of scope of what networkd would do?